src/SUFST/src/can_unpack.c

```diff
@@ -77,38 +113,31 @@ void queue_receive_thread_entry(ULONG input)
               break;
             }
             /* Couldn't find matching identifier - deassign pointer. */
-            if(i == TABLE_SIZE)handlerunpack = NULL;
+            if(id == TABLE_SIZE)handlerunpack = NULL;
         }
         /* Skip frame if couldn't find matching identifier. */
         if(handlerunpack == NULL)
         {
           continue;
         }
         /* Check latest timestamp in ts_table, skip frame if not enough time has elapsed. Update ts_table. */
-        l_timestamp = ts_table[i];
+        /* This part will not be needed when this feature will be implemented in rtcan */
+        l_timestamp = ts_table[id];
         c_timestamp = tx_time_get();
         if (c_timestamp - l_timestamp < 50)
         {
           continue;
         }
-        ts_table[i] = c_timestamp;
+        ts_table[id] = c_timestamp;
 
         /* Fill pdu_struct data buffer */
         handlerunpack->unpack_func((uint8_t *) &pdu_struct.data, rx_msg.data, rx_msg.length);
 
         pdu_struct.header.epoch = c_timestamp; /* Assign timestamp */
         pdu_struct.start_byte = 1; /* Assign start byte */
-        pdu_struct.ID = 0; /* Assign PDU ID */
+        pdu_struct.ID = id; /* Assign PDU ID */
```

Comment:
Does the order of PDU IDs in the telemetry servers schema match the order of PDU IDs in the generated table of handlers? If it does this is fine for now, but we definitely need to develop a system for setting the IDs properly (currently they're just assigned in the order that the telemetry generator script reads through the CAN database).

Reply:
Yes, schema IDs matches order of PDUs.